format/src/main/scala/org/scalafmt/internal/TokenList.scala

```diff
@@ -0,0 +1,40 @@
+package org.scalafmt.internal
+
+import scala.collection.SeqView
+import scala.collection.immutable.IndexedSeq
+import scala.meta.tokens.Token
+import scala.meta.tokens.Tokens
+
+/** Helper to traverse tokens as a doubly linked list.  */
+final class TokenList private (tokens: Tokens) {
```

Comment:
Given that all methods are stateless thanks to binarySearch here we can remove `TokenList` and move the utilites to the extension methods